scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3080,18 +3083,18 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
     trees
   }
 
-  def blockExpr(): Term = autoPos {
+  def blockExpr(isAfterArrow: Boolean = false): Term = autoPos {
     if (ahead(token.is[CaseIntro] || (token.is[Ellipsis] && ahead(token.is[KwCase])))) {
       if (token.is[LeftBrace])
         Term.PartialFunction(inBraces(caseClauses()))
       else
         Term.PartialFunction(indented(caseClauses()))
-    } else block()
+    } else block(isAfterArrow)
   }
 
-  def block(): Term = autoPos {
+  def block(isAfterArrow: Boolean = false): Term = autoPos {
     def blockWithStats = Term.Block(blockStatSeq())
-    if (token.is[LeftBrace]) {
+    if (!isAfterArrow && token.is[LeftBrace]) {
```

Comment:
should we rename this new parameter based on what we're doing with it here, not where it came from? what if another case could lead to this as well?

Reply:
This is just the situation where {} is optional for blocks, which is only after `=>`. I think the name makes sense here.

Reply:
so why not rename `isBlockOptional`?

Reply:
changed!